Update documentation (#24)
Consensys/linea-sequencer: PLUGINS.md

The profitability calculator is a shared component, that is used to check if a tx is profitable.
It is applied, with different configuration to:
1. `linea_estimateGas` endpoint
2. Tx validation for the txpool (if tx profitability check is enabled)
3. Tx selection during block creation

#### CLI Options

| Command Line Argument                                    | Default Value |
|----------------------------------------------------------|---------------|
| `--plugin-linea-fixed-gas-cost-wei`                      | 0             |
| `--plugin-linea-variable-gas-cost-wei`                   | 1_000_000_000 |
| `--plugin-linea-extra-data-pricing-enabled`              | false         |
| `--plugin-linea-min-margin`                              | 1.0           |
| `--plugin-linea-estimate-gas-min-margin`                 | 1.0           |
| `--plugin-linea-tx-pool-min-margin`                      | 0.5           |

### Module line count validator

> The Module line count validator is a shared component, that is used to check if a tx exceed any of the configured line count limit.

Reviewer: ```suggestion
The Module line count validator is a shared component, that is used to check if a tx exceeds any of the configured line count limits.
```

It is used in:
1. `linea_estimateGas` endpoint
2. Tx validation for the txpool (if tx simulation is enabled)
3. Tx selection during block creation

#### CLI Options

| Command Line Argument                                 | Default Value        |
|-------------------------------------------------------|----------------------|
| `--plugin-linea-module-limit-file-path`               | moduleLimitFile.toml |
| `--plugin-linea-over-line-count-limit-cache-size`     | 10_000               |

### L1 L2 Bridge

> These value are just passed to the ZkTracer

Reviewer: ```suggestion
These values are just passed to the ZkTracer
```

This includes setting limits such as `TraceLineLimit`, `maxBlockGas`, and `maxCallData`, and check the profitability
of a transaction.
The selectors are in the package `net.consensys.linea.sequencer.txselection.selectors`.

#### CLI Options

| Command Line Argument                                  | Default Value        |
|--------------------------------------------------------|----------------------|
| `--plugin-linea-max-block-calldata-size`               | 70000                |
| `--plugin-linea-module-limit-file-path`                | moduleLimitFile.toml |
| `--plugin-linea-over-line-count-limit-cache-size`      | 10_000               |
| `--plugin-linea-max-block-gas`                         | 30_000_000L          |
| `--plugin-linea-unprofitable-cache-size`               | 100_000              |
| `--plugin-linea-unprofitable-retry-limit`              | 10                   |

### Transaction Validation - LineaTransactionValidatorPlugin

This plugin extends the default transaction validation rules for adding transactions to the
transaction pool. It leverages the `PluginTransactionValidatorService` to manage and customize the
process of transaction validation.

> This includes setting limits such as `TraceLineLimit`, `maxTxGasLimit`, and `maxTxCallData`, and check the profitability

Reviewer: ```suggestion
This includes setting limits such as `TraceLineLimit`, `maxTxGasLimit`, and `maxTxCallData`, and checking the profitability
```

|----------------------------------------------------------|-------------------|
| `--plugin-linea-deny-list-path`                          | lineaDenyList.txt |
| `--plugin-linea-max-tx-gas-limit`                        | 30_000_000        |
| `--plugin-linea-max-tx-calldata-size`                    | 60_000            |
| `--plugin-linea-tx-pool-simulation-check-api-enabled`    | false             |
| `--plugin-linea-tx-pool-simulation-check-p2p-enabled`    | false             |
| `--plugin-linea-tx-pool-profitability-check-api-enabled` | true              |
| `--plugin-linea-tx-pool-profitability-check-p2p-enabled` | false             |

## RPC Methods

### Linea Estimate Gas
#### `linea_estimateGas`

This endpoint simulates a transaction, including line count limit validation, and returns the estimated gas used ( as the standard `eth_estimateGas`) plus the estimated gas price to be used when submitting the tx. 

#### Parameters
same as `eth_estimateGas`

#### Result
```json
{
  "jsonrpc": "2.0",
  "id": 53,
  "result": {
    "gasLimit": "0x5208",
    "baseFeePerGas": "0x7",
    "priorityFeePerGas": "0x123456"
  }
}
```

### Linea Set Extra Data
#### `linea_setExtraData`

> This endpoint is used to configure the extra data based pricing, and only make sense to call it on the sequencer.

Reviewer: ```suggestion
This endpoint is used to configure the extra data based pricing, and it only makes sense to call it on the sequencer.
```
